src/rt/lifetime.d

```diff
@@ -1193,24 +1186,53 @@ extern (C) void _d_delarray_t(void[]* p, const TypeInfo_Struct ti)
 {
     if (p)
     {
-        assert(!(*p).length || (*p).ptr);
-        if ((*p).ptr)
+        auto bic = __getBlkInfo(p.ptr);
+        auto info = bic ? *bic : GC.query(p.ptr);
+
+        if (info.base && (info.attr & BlkAttr.APPENDABLE))
```

Comment:
I don't know whether it's expected or not, but I think I might expect either possibility when I call delete.

What does make the most sense to me is to have delete be a "pre-emptive GC collection," so everything that is done via GC would be done via delete, except synchronously.

In any case, we are spending time debating a to-be-deprecated feature :) Nice to have these discussions though, for when we do actually have to create a replacement.